### PR TITLE
Docker support and deployment workflow

### PR DESCRIPTION
**Pull request by BingoDeeVee**
Includes a `dockerfile` and an actions workflow for deploying to DockerHub. Also updates `.gitignore` for the benefit of pipenv users looking to contribute.

### CODE REVIEW

File: .github/workflows/main_snitching-confessions-bot.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PAT }}
+      -
+        name: Build and push
+        uses: docker/build-push-action@v6
+        with:
+          push: true
+          tags: bingodeevee/snitching-confessions-dockerfork:latest
```
Comment (bbaster) [inline .github/workflows/main_snitching-confessions-bot.yml:60]: Line 60 references your fork

Comment (bbaster) [reply .github/workflows/main_snitching-confessions-bot.yml:60]: Line 33 also

Comment (bbaster) [reply .github/workflows/main_snitching-confessions-bot.yml:60]: And is QEMU necessary?

**Review by bbaster — CHANGES_REQUESTED** [review]

File: .github/workflows/main_snitching-confessions-bot.yml (hunk 1 above)
Comment (BingoDeeVee) [reply .github/workflows/main_snitching-confessions-bot.yml:60]: > Line 60 references your fork

Those are docker tag references for dockerhub. Will need to change the name soon to align with the repo name change but it's a bigger piece of work so I've kept it the same.

Comment (BingoDeeVee) [reply .github/workflows/main_snitching-confessions-bot.yml:60]: > And is QEMU necessary?

Yes, the build runner can't invoke docker without it.

**Review by bbaster — APPROVED** [review]

**Pull request merged**